### PR TITLE
POC: Parallel builds

### PR DESCRIPTION
**Pull request by kevincolten**
These tweaks allow you to enable parallel builds with https://github.com/trivago/parallel-webpack

You will have to manually separate your packs in your `config/webpack/environment.js`:
```js
const clone = require('clone');
const { environment } = require('@rails/webpacker');

module.exports = Object.keys(environment.entry).map(key => {
  const bundle = clone(environment);
  bundle.entry = {
    [key]: environment.entry[key]
  }
  return bundle;
})
```

and in your individual environment files (eg. `config/webpack/production.js`)
```js
const environment = require('./environment');
const { parse } = require('path');

module.exports = environment.map(env => {
  const path = `app/javascript/packs/${parse(Object.values(env.entry)[0]).base}`;
  return env.toWebpackConfig(path);
});
```

### CODE REVIEW

File: package/environments/base.js [hunk 1]
```diff
@@ -111,7 +112,12 @@ module.exports = class Base {
     this.resolvedModules = getModulePaths()
   }
 
-  toWebpackConfig() {
+  toWebpackConfig(path) {
+    this.loaders = getLoaderList()
```
Comment (kevincolten) [inline package/environments/base.js]: Had to reset these properties here. Something to do with the parallelization was stripping them of being ConfigObjects

**Comment by guilleiguaran** [comment]
This is causing CI failures 😬

**Comment by kevincolten** [comment]
Yeah, I’ll take a look. Wanted to gauge interest first

**Comment by javan** [comment]
I'm hesitant to make changes just to support a 3rd-party webpack plugin. Could it be reframed as a more generalized feature like "Add support for compiling a single pack"?

**Comment by drewB** [comment]
Bummed this didn't move forward.  I would hope would help with things like https://github.com/rails/webpacker/issues/1347

**Pull request closed without merging**